src/ol_infrastructure/components/services/k8s.py

```diff
@@ -0,0 +1,311 @@
+                                        container_port=DEFAULT_UWSGI_PORT
+                                    )
+                                ],
+                                image_pull_policy="IfNotPresent",
+                                resources=kubernetes.core.v1.ResourceRequirementsArgs(
+                                    requests={"cpu": "250m", "memory": "300Mi"},
+                                    limits={"cpu": "500m", "memory": "600Mi"},
+                                ),
+                                env=application_deployment_env_vars,
+                                env_from=application_deployment_envfrom,
+                                volume_mounts=[
+                                    kubernetes.core.v1.VolumeMountArgs(
+                                        name="staticfiles",
+                                        mount_path="/src/staticfiles",
+                                    ),
+                                ],
+                            ),
+                        ],
+                    ),
+                ),
+            ),
+            opts=resource_options,
+        )
+
+        # A kubernetes service resource to act as load balancer for the app instances
+        _application_service = kubernetes.core.v1.Service(
+            f"unified-application-{stack_info.env_suffix}-service",
+            metadata=kubernetes.meta.v1.ObjectMetaArgs(
+                name=self.ol_app_k8s_config.application_lb_service_name,
+                namespace=self.ol_app_k8s_config.application_namespace,
+                labels=self.ol_app_k8s_config.k8s_global_labels,
+            ),
+            spec=kubernetes.core.v1.ServiceSpecArgs(
+                selector=application_labels,
+                ports=[
+                    kubernetes.core.v1.ServicePortArgs(
+                        name=self.ol_app_k8s_config.application_lb_service_name,
+                        port=DEFAULT_NGINX_PORT,
+                        target_port=DEFAULT_NGINX_PORT,
+                        protocol="TCP",
+                    ),
+                ],
+                type="ClusterIP",
+            ),
+            opts=resource_options,
+        )
+
+        _application_pod_security_group_policy = (
+            kubernetes.apiextensions.CustomResource(
+                f"unified-application-{stack_info.env_suffix}-application-pod-security-group-policy",
+                api_version="vpcresources.k8s.aws/v1beta1",
+                kind="SecurityGroupPolicy",
+                metadata=kubernetes.meta.v1.ObjectMetaArgs(
+                    name="application-app",
+                    namespace=self.ol_app_k8s_config.application_namespace,
+                    labels=self.ol_app_k8s_config.k8s_global_labels,
+                ),
+                spec={
+                    "podSelector": {
+                        "matchLabels": {
+                            "ol.mit.edu/pod-security-group": "application-app"
```

Comment:
pod-security-group labels should use the `ol_app_k8s_config.application_name` value as well.